core/modules/url_filter.h

```diff
@@ -17,44 +17,44 @@
 
 using bess::utils::TcpFlowReconstruct;
 using bess::utils::Trie;
+using bess::utils::be16_t;
+using bess::utils::be32_t;
 
 // A helper class that defines a TCP flow
 class alignas(16) Flow {
  public:
-  union {
-    struct {
-      uint32_t src_ip;
-      uint32_t dst_ip;
-      uint16_t src_port;
-      uint16_t dst_port;
-    };
-
-    struct {
-      uint64_t e1;
-      uint64_t e2;
-    };
-  };
-
-  Flow() : e1(0), e2(0) {}
+  be32_t src_ip;
+  be32_t dst_ip;
+  be16_t src_port;
+  be16_t dst_port;
+  uint32_t padding;
+
+  Flow() : padding(0) {}
 
   bool operator==(const Flow &other) const {
-    return e1 == other.e1 && e2 == other.e2;
+    return memcmp(this, &other, sizeof(*this)) == 0;
   }
 };
 
+static_assert(sizeof(Flow) == 16, "Flow must be 16 bytes.");
+
 // Hash function for std::unordered_map
 struct FlowHash {
   std::size_t operator()(const Flow &f) const {
-    static_assert(sizeof(Flow) == 2 * sizeof(uint64_t),
```

Comment:
Don't we want to keep the static assert to ensure that the flow struct is exactly 16 bytes?